yt_dlp/extractor/nhk.py

```diff
@@ -591,102 +592,198 @@ class NhkRadiruIE(InfoExtractor):
     }, {
         # news
-        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=F261_01_4012173',
+        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=F261_01_4178426',
+        'skip': 'Expires on 2025-03-23',
         'info_dict': {
-            'id': 'F261_01_4012173',
+            'id': 'F261_01_4178426',
             'ext': 'm4a',
-            'channel': 'NHKラジオ第1',
+            'title': '午前８時のNHKニュース',
             'uploader': 'NHKラジオ第1',
-            'series': 'NHKラジオニュース',
-            'title': '午前０時のNHKニュース',
+            'channel': 'NHKラジオ第1',
             'thumbnail': 'https://www.nhk.or.jp/radioondemand/json/F261/img/RADIONEWS_640.jpg',
-            'release_timestamp': 1718290800,
-            'release_date': '20240613',
-            'timestamp': 1718291400,
-            'upload_date': '20240613',
+            'series': 'NHKラジオニュース',
+            'timestamp': 1742684700,
+            'upload_date': '20250322',
+            'release_timestamp': 1742684400,
+            'release_date': '20250322',
+
         },
     }, {
         # fallback when extended metadata fails
-        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=2834_01_4009298',
-        'skip': 'Expires on 2024-06-07',
+        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=J8792PY43V_14_4177121',
+        'skip': 'Expires on 2025-04-14',
         'info_dict': {
-            'id': '2834_01_4009298',
-            'title': 'まち☆キラ！開成町特集',
+            'id': 'J8792PY43V_14_4177121',
             'ext': 'm4a',
-            'release_date': '20240531',
-            'upload_date': '20240531',
-            'series': 'はま☆キラ！',
-            'thumbnail': 'https://www.nhk.or.jp/prog/img/2834/g2834.jpg',
-            'channel': 'NHK R1,FM',
-            'description': '',
-            'timestamp': 1717123800,
-            'uploader': 'NHK R1,FM',
-            'release_timestamp': 1717120800,
-            'series_id': '2834_01',
+            'title': '「白い蝶［ちょう］を探してみよう」いきもの☆いろいろ',
+            'description': '海野 和男（昆虫写真家）',
+            'uploader': 'NHK R1',
+            'channel': 'NHK R1',
+            'thumbnail': 'https://www.nhk.or.jp/radioondemand/json/J8792PY43V/img/corner/box_26_thumbnail.jpg',
+            'series': 'マイあさ！ いきもの☆いろいろ',
+            'series_id': 'J8792PY43V_14',
+            'timestamp': 1741985700,
+            'upload_date': '20250314',
+            'release_timestamp': 1741982400,
+            'release_date': '20250314',
         },
-        'expected_warnings': ['Failed to get extended metadata. API returned empty list.'],
+        'expected_warnings': ['Failed to download extended metadata: HTTP Error 404: Not Found'],
     }]
 
     _API_URL_TMPL = None
 
+    # the following few functions are ported from https://www.nhk.or.jp/radio/assets/js/timetable_detail_new.js
+
+    def _format_actlist(self, act_list):
```

Comment:
more readable with the underscore IMO
```suggestion
    def _format_act_list(self, act_list):
```